### PR TITLE
Clean up some robot code

### PR DESCRIPTION
**Pull request by flrsh**
[CLEAN] [BUG]

## About the PR 

This PR cleans up some robot code and refactors things a bit to use the very useful find_ghost_by_key() proc. It also fixes a bug where borgs falling into a pile of components wouldn't actually drop any of their components.

## Why's this needed? 

Clean good, bugs bad.

### CODE REVIEW

**Review by ZeWaka — APPROVED** [review]

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -1136,17 +1136,17 @@
...
+					if (!M) // if we couldn't find them (i.e. they're still alive), don't pull them into this borg
+						src.visible_message("<span class='alert'><b>[src]</b> remains inactive.</span>")
+						return
+					if (!isdead(M)) // so if they're in VR, the afterlife bar, or a ghostcritter
+						boutput(M, "<span class='notice'>You feel yourself being pulled out of your current plane of existence!</span>")
+						B.owner = M.ghostize()?.mind
+						qdel(M)
 					B.owner.transfer_to(src)
+					if (B.owner.current.client)
+						src.lastKnownIP = B.owner.current.client.address
```
Comment (MarkNais) [inline code/mob/living/silicon/robot.dm:1149]: This could maybe be in the `transfer_to()` proc

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -2891,6 +2891,20 @@
 	proc/compborg_take_critter_damage(var/zone = null, var/brute = 0, var/burn = 0)
 		TakeDamage(pick(get_valid_target_zones()), brute, burn)
 
+	proc/collapse_to_pieces()
+		src.visible_message("<span class='alert'><b>[src]</b> falls apart into a pile of components!</span>")
+		var/turf/T = get_turf(src)
+		for(var/obj/item/parts/robot_parts/R in src.contents)
+			R.set_loc(T)
```
Comment (MarkNais) [inline code/mob/living/silicon/robot.dm:2898]: I'm unfamiliar with robot code. Would 100% damaged (utterly destroyed) limbs still be in contents?

Comment (MarkNais) [reply code/mob/living/silicon/robot.dm:2898]: For that matter, do drop limbs store damage? (pretty sure not) However breaking down a robot in order to repair it magically seems an odd edge case we might not need to worry about.

**Review by MarkNais — APPROVED** [review]
Seems good. Was it tested?

File: code/mob/living/silicon/robot.dm (hunk 1 above)
Comment (flrsh) [reply code/mob/living/silicon/robot.dm:1149]: Fair! I'm not refactoring all the other instances that also do this though :sheltersweat:

File: code/mob/living/silicon/robot.dm (hunk 2 above)
Comment (flrsh) [reply code/mob/living/silicon/robot.dm:2896]: > I'm unfamiliar with robot code. Would 100% damaged (utterly destroyed) limbs still be in contents?

They would not!

**Comment by flrsh** [comment]
> Seems good. Was it tested?

Yep!

**Pull request merged**